File: frontend/src/pages/Teachers/teacherService.js
import api from './api';

const teacherService = {

  getAll: async (page = 0, size = 10) => {
    const response = await api.get('/api/teachers', {
      params: {
        page,
        size
      }
    });

    return response.data;
  },

  getById: async (id) => {
    const response = await api.get(`/api/teachers/${id}`);
    return response.data;
  },

  create: async (data) => {
    const response = await api.post('/api/teachers', data);
    return response.data;
  },

  update: async (id, data) => {
    const response = await api.put(`/api/teachers/${id}`, data);
    return response.data;
  },

  remove: async (id) => {
    const response = await api.delete(`/api/teachers/${id}`);
    return response.data;
  }

};

export default teacherService;

File: backend-java/src/main/java/com/example/faceattendance/service/TeacherService.java
package com.example.faceattendance.service;

import com.example.faceattendance.dto.teacher.CreateTeacherRequest;
import com.example.faceattendance.dto.teacher.TeacherResponse;
import com.example.faceattendance.dto.teacher.UpdateTeacherRequest;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

/**
 * Teacher management service contract.
 */
public interface TeacherService {

    TeacherResponse create(CreateTeacherRequest request);

    Page<TeacherResponse> getAll(Pageable pageable, String search);

    TeacherResponse getById(Long id);

    TeacherResponse update(Long id, UpdateTeacherRequest request);

    void delete(Long id);

}